app/src/main/java/org/systers/mentorship/view/activities/MemberProfileActivity.kt

```diff
@@ -49,10 +49,19 @@ class MemberProfileActivity : BaseActivity() {
         memberProfileViewModel.getUserProfile(userId)
 
         btnSendRequest.setOnClickListener {
-            val intent = Intent(this@MemberProfileActivity, SendRequestActivity::class.java)
-            intent.putExtra(SendRequestActivity.OTHER_USER_ID_INTENT_EXTRA, userProfile.id)
-            intent.putExtra(SendRequestActivity.OTHER_USER_NAME_INTENT_EXTRA, userProfile.name)
-            startActivity(intent)
+
+            if (memberProfileViewModel.userProfile.isAvailableToMentor!!
+                    || memberProfileViewModel.userProfile.needsMentoring!!) {
+
+                val intent = Intent(this@MemberProfileActivity, SendRequestActivity::class.java)
+                intent.putExtra(SendRequestActivity.OTHER_USER_ID_INTENT_EXTRA, userProfile.id)
+                intent.putExtra(SendRequestActivity.OTHER_USER_NAME_INTENT_EXTRA, userProfile.name)
+                startActivity(intent)
+
+            } else if (!memberProfileViewModel.userProfile.needsMentoring!!
```

Comment:
This `else if` statement is the opposite of the `if` statement, so it makes sense to just make it an `else` statement